matcher/src/chars/normalize.rs

```diff
@@ -1,526 +1,972 @@
+    'U', // 'Ự'; '\u{1ef0}'
+    'u', // 'ự'; '\u{1ef1}'
+    'Y', // 'Ỳ'; '\u{1ef2}'
+    'y', // 'ỳ'; '\u{1ef3}'
+    'Y', // 'Ỵ'; '\u{1ef4}'
+    'y', // 'ỵ'; '\u{1ef5}'
+    'Y', // 'Ỷ'; '\u{1ef6}'
+    'y', // 'ỷ'; '\u{1ef7}'
+    'Y', // 'Ỹ'; '\u{1ef8}'
+    'y', // 'ỹ'; '\u{1ef9}'
+    'Ỻ', // 'Ỻ'; '\u{1efa}'
+    'ỻ', // 'ỻ'; '\u{1efb}'
+    'Ỽ', // 'Ỽ'; '\u{1efc}'
+    'ỽ', // 'ỽ'; '\u{1efd}'
+    'Ỿ', // 'Ỿ'; '\u{1efe}'
+    'ỿ', // 'ỿ'; '\u{1eff}'
 ];
 
-const DATA3: [(char, char); 9] = [
-    ('\u{2071}', 'i'), // , SUPERSCRIPT LATIN SMALL LETTER
-    ('\u{2095}', 'h'), // , LATIN SUBSCRIPT SMALL LETTER
-    ('\u{2096}', 'k'), // , LATIN SUBSCRIPT SMALL LETTER
-    ('\u{2097}', 'l'), // , LATIN SUBSCRIPT SMALL LETTER
-    ('\u{2098}', 'm'), // , LATIN SUBSCRIPT SMALL LETTER0x2099: 'n', // , LATIN SUBSCRIPT SMALL LETTER
-    ('\u{209A}', 'p'), // , LATIN SUBSCRIPT SMALL LETTER
-    ('\u{209B}', 's'), // , LATIN SUBSCRIPT SMALL LETTER
-    ('\u{209C}', 't'), // , LATIN SUBSCRIPT SMALL LETTER
-    ('\u{2184}', 'c'), // , LATIN SMALL LETTER REVERSED
+/// A char array corresponding to the following Unicode block:
+///
+/// - [Superscripts and Subscripts](https://en.wikipedia.org/wiki/Superscripts_and_Subscripts)
+///
+/// This covers the range `'\u{2070}'..='\u{209f}'`.
+const SUPERSCRIPTS_AND_SUBSCRIPTS: [char; 48] = [
+    '0', // '⁰'; '\u{2070}'
+    'i', // 'ⁱ'; '\u{2071}'
+    '⁲', // '⁲'; '\u{2072}'
+    '⁳', // '⁳'; '\u{2073}'
+    '4', // '⁴'; '\u{2074}'
+    '5', // '⁵'; '\u{2075}'
+    '6', // '⁶'; '\u{2076}'
+    '7', // '⁷'; '\u{2077}'
+    '8', // '⁸'; '\u{2078}'
+    '0', // '⁹'; '\u{2079}'
+    '+', // '⁺'; '\u{207a}'
+    '-', // '⁻'; '\u{207b}'
+    '=', // '⁼'; '\u{207c}'
+    '(', // '⁽'; '\u{207d}'
+    ')', // '⁾'; '\u{207e}'
+    'n', // 'ⁿ'; '\u{207f}'
+    '0', // '₀'; '\u{2080}'
+    '1', // '₁'; '\u{2081}'
+    '2', // '₂'; '\u{2082}'
+    '3', // '₃'; '\u{2083}'
+    '4', // '₄'; '\u{2084}'
+    '5', // '₅'; '\u{2085}'
+    '6', // '₆'; '\u{2086}'
+    '7', // '₇'; '\u{2087}'
+    '8', // '₈'; '\u{2088}'
+    '9', // '₉'; '\u{2089}'
+    '+', // '₊'; '\u{208a}'
+    '-', // '₋'; '\u{208b}'
+    '=', // '₌'; '\u{208c}'
+    '(', // '₍'; '\u{208d}'
+    ')', // '₎'; '\u{208e}'
+    '₏', // '₏'; '\u{208f}'
+    'a', // 'ₐ'; '\u{2090}'
+    'e', // 'ₑ'; '\u{2091}'
+    'o', // 'ₒ'; '\u{2092}'
+    'x', // 'ₓ'; '\u{2093}'
+    'e', // 'ₔ'; '\u{2094}'
+    'h', // 'ₕ'; '\u{2095}'
+    'k', // 'ₖ'; '\u{2096}'
+    'l', // 'ₗ'; '\u{2097}'
+    'm', // 'ₘ'; '\u{2098}'
+    'n', // 'ₙ'; '\u{2099}'
+    'p', // 'ₚ'; '\u{209a}'
+    's', // 'ₛ'; '\u{209b}'
+    't', // 'ₜ'; '\u{209c}'
+    '₝', // '₝'; '\u{209d}'
+    '₞', // '₞'; '\u{209e}'
+    '₟', // '₟'; '\u{209f}'
 ];
 
-const DATA1_START: u32 = DATA1[0].0 as u32;
-const DATA1_END: u32 = DATA1[DATA1.len() - 1].0 as u32 + 1;
-const LEN1: usize = (DATA1_END - DATA1_START) as usize;
-static TABLE1: [char; LEN1] = generate_table(&DATA1);
+#[cfg(test)]
+mod tests {
```

Comment:
thanks for adding some tests, definitely needed here!

Reply:
No problem!